src/shims/index.js

```diff
@@ -1,9 +1,34 @@
+// KEEP THIS AS NON-ES-6 MODULE so build-shims.js can use it
+
+exports.SHIMMED_MODULES = [
+	'preact',
+	'preact-router',
+	'preact-compat',
+	'preact-redux',
+	'react-apollo'
+];
+
 /**
  * A shimmed module is injected into the global object by the parent app.
  * The shimmed module is then aliased as the real module, and Zimlet consumers
  * will not need to duplicate libraries used by the main app.
  */
 
-export function getShimPath(module) {
+exports.getShimPath =  function(module) {
 	return require.resolve(`./${module}`);
-}
+};
+
+/**
+ * Modify the shimmed module function to throw a meaningful error message
+ * in the event that a method we think should be provided by the module is not actually
+ * available and is attempted to be used
+ * @param {String} moduleName The name of the module to be used in the error log
+ * @param {Object} e the `exports` object from a module
+ */
+exports.warnOnMissingExport = function(m, moduleName, namedExport) {
+	return m && typeof m[namedExport] !== 'undefined' ?
```

Comment:
This will guard against the shim not being passed down at all from zm-x-web as well as the named export not existing!